### PR TITLE
feat(catalog): Remove temporary caching in favour of stale-while-revalidate

### PR DESCRIPTION
**Pull request by benjdlambert**
So, here's a nice one.

I was browsing around and looking at solutions, before the best i had was mentioned in #1230.

Tonight I picked this up and found a really nice hook from the guys over at `vercel` / `zeit`. If you wanna read more about it, you can check out how much this library can do here: https://swr.now.sh/.

I think that it actually might be a good idea to promote this hook in the boilerplate, as it's not just data fetching that it can be applied to, it's basically anything you always want to return a value for, but then maybe change that when the processing/waiting is done.

It's a super clean API that provides the ability to hook into a lot of the async lifecycle, things like what to do if this request takes too long, or if the user comes back to the tab after being inactive, or one of my personal favourites that I have always found is an absolute PITA is loading data and keeping the scroll position: https://swr.now.sh/#scroll-position-recovery-and-pagination. (Srsly, just this.)

It supports also `suspense` which would be super nice to include to improve the end user experience for loading async data or doing async stuff further down the line.

For the record, I'd really like to suggest that if you don't use GraphQL later on down the line, (which hopefully would support `suspense`) that you should use this hook to fetch your data, as it gives you everything you all the things you need to make a great user experience.

If you can't tell, I think I've actually fallen for this library, it seems to fix a lot of stuff that I've had the worst experience with over the years.

#### :heavy_check_mark: Checklist

- [x] All tests are passing `yarn test`
- [ ] Screenshots attached (for UI changes)
- [ ] Relevant documentation updated
- [ ] Prettier run on changed files
- [ ] Tests added for new functionality
- [ ] Regression tests added for bug fixes

### CODE REVIEW

File: plugins/catalog/src/components/CatalogPage/CatalogPage.tsx [hunk 1]
```diff
@@ -45,6 +44,7 @@ import {
   CatalogFilterItem,
 } from '../CatalogFilter/CatalogFilter';
 import { CatalogTable } from '../CatalogTable/CatalogTable';
+import useStaleWhileRevalidate from 'swr';
```
Comment (Rugvip) [inline plugins/catalog/src/components/CatalogPage/CatalogPage.tsx:47]: Not useSWR? :grin:

**Review by Rugvip — APPROVED** [review]
Noice, +1 to highlighting if it ends up working well.

Assuming the scope of the keys is global, which could potentially lead to trouble in the long run. Maybe a cache override could be added to the config context so that we could wrap our plugins in separate SWR configs.

File: plugins/catalog/src/components/CatalogPage/CatalogPage.tsx (hunk 1 above)
Comment (benjdlambert) [reply plugins/catalog/src/components/CatalogPage/CatalogPage.tsx:47]: When I saw the hook for the first time, I had no idea what was going on. So I thought I'd try to call it something a little more semantic. haha. 

https://en.wikipedia.org/wiki/SWR

Comment (Rugvip) [reply plugins/catalog/src/components/CatalogPage/CatalogPage.tsx:47]: Yeah true, although tbh I don't think StaleWhileRevalidate is too obvious either x)

Comment (benjdlambert) [reply plugins/catalog/src/components/CatalogPage/CatalogPage.tsx:47]: `useReallyGoodCachyLibraryThing`

Comment (Rugvip) [reply plugins/catalog/src/components/CatalogPage/CatalogPage.tsx:47]: useRGCLT

Comment (benjdlambert) [reply plugins/catalog/src/components/CatalogPage/CatalogPage.tsx:47]: That's what I'm talking about.

Comment (benjdlambert) [reply plugins/catalog/src/components/CatalogPage/CatalogPage.tsx:47]: Nailed it

**Pull request merged**